app/controllers/settings_controller.rb

```diff
@@ -22,14 +22,15 @@ def update_ldap
   end
 
   def update_profile
-    @current_user.group_users.destroy_all
-    @current_user.update(user_params)
+    unless @current_user.update(user_params)
```

Comment:
Favor modifier `unless` usage when having a single-line body. Another good alternative is the usage of control flow `&&`/`||`.